openhands/server/routes/secrets.py

```diff
@@ -75,7 +75,8 @@ async def check_provider_tokens(
     if incoming_provider_tokens.provider_tokens:
         # Determine whether tokens are valid
         for token_type, token_value in incoming_provider_tokens.provider_tokens.items():
-            if token_value.token:
+            # Only validate if token is not empty
+            if token_value.token and token_value.token.get_secret_value():
```

Comment:
```suggestion
            if token_value.token:
```

We have many cases where the agent attempts to extract the secret value; pydantic will evaluate this as False is the value is `SecretStr('')` without use have to extract the value